### PR TITLE
feat: removing user from organisation

### PR DESCRIPTION
**Pull request by MuthoniMN**
## What does this PR do?
This PR aims to allow an organisation owner to remove a member from their organisation. 

## How can this be manually tested?
1. Create an organisation using POST `/api/v1/organisations`
2. Create a new user and get their user ID
3. Add the user to the organisation
4. Remove the user from the organisation

## Edge Cases
- Test when the organisation does not exist
- Test when the current user is not the owner of the organisation
- Test when the specified user is not a member of the organisation

## Checklist
- [x] Create the endpoint HTTP DELETE `/api/v1/organisations/{org_id}/users/{user_id}` to remove a user from an organization.
- [x] Implement authentication middleware to validate JWT tokens.
- [x] Implement an authorization check to ensure that only the organization's owner has the ability to remove users.
- [x] Validate the request param {orgId} and {userId}
- [x] Write unit tests for all scenarios, including successful removal of user from organization and validation errors.

### Successful Response
```
{
      status: 'Success',
      message: 'User removed successfully',
      status_code: 200,
}
```
### Error Response when organisation does not exist
```
{
   status: 'error',
   message: 'Organisation not found',
    status_code: 404,
}
```
### Error Response when the current user does not own the organisation
```
{
        status: 'Forbidden',
        message: 'Only admin can remove users',
        status_code: 403,
}
```
### Error Response when the specified user is not a member of the organisation
```
{
        status: 'error',
        message: 'User not found',
        status_code: 404,
}
```
## Associated Issue
Issue #61

### CODE REVIEW

**Review by Heba-WebDev — DISMISSED** [review]
All good. Good job!

**Review by davidolanrewaju — APPROVED** [review]

File: src/modules/organisations/organisations.controller.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Body, Controller, Param, Patch, Post, Request } from '@nestjs/common';
+import { Body, Controller, Param, Patch, Post, Request, Delete } from '@nestjs/common';
 import { OrganisationsService } from './organisations.service';
```
Comment (davidolanrewaju) [inline src/modules/organisations/organisations.controller.ts:2]: Remove all methods you are not using from your import

**Comment by davidolanrewaju** [comment]
Please fix your workflow, pull the latest changes to the repo run your install and push back

**Review by davidolanrewaju — DISMISSED** [review]

**Review by Heba-WebDev — APPROVED** [review]

File: src/modules/organisations/organisations.service.ts
Comment (Heba-WebDev) [inline src/modules/organisations/organisations.service.ts:1]: Remove user method is doing so much. Extract these helper methods and just call them in remove user. Let's follow what buka advice us to do, each method does 1 single thing.

Comment (MuthoniMN) [reply src/modules/organisations/organisations.service.ts:1]: I've created helper methods. You can review again. Thank you

And I've ensured the tests are successful as well

**Review by Homoakin619 — DISMISSED** [review]

File: src/modules/organisations/organisations.service.ts
Comment (Heba-WebDev) [inline src/modules/organisations/organisations.service.ts:1]: Please change the name to removeMember, since these are not users but rather members of the organization

**Comment by Heba-WebDev** [comment]
Merge conflicts, please resolve

File: src/modules/organisations/organisations.service.ts
Comment (MuthoniMN) [reply src/modules/organisations/organisations.service.ts:1]: Okay. I've updated

**Review by Heba-WebDev — DISMISSED** [review]

**Review by Heba-WebDev — APPROVED** [review]

**Review by Cobidex — DISMISSED** [review]
